src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/UseSpanClearInsteadOfFill.cs

```diff
@@ -0,0 +1,155 @@
+        private static readonly LocalizableString s_localizableMessage = new LocalizableResourceString(nameof(MicrosoftNetCoreAnalyzersResources.UseSpanClearInsteadOfFillMessage), MicrosoftNetCoreAnalyzersResources.ResourceManager, typeof(MicrosoftNetCoreAnalyzersResources));
+        private static readonly LocalizableString s_localizableDescription = new LocalizableResourceString(nameof(MicrosoftNetCoreAnalyzersResources.UseSpanClearInsteadOfFillDescription), MicrosoftNetCoreAnalyzersResources.ResourceManager, typeof(MicrosoftNetCoreAnalyzersResources));
+
+        internal static readonly DiagnosticDescriptor s_Rule = DiagnosticDescriptorHelper.Create(
+            DiagnosticId,
+            s_localizableTitle,
+            s_localizableMessage,
+            DiagnosticCategory.Performance,
+            RuleLevel.BuildWarning,
+            description: s_localizableDescription,
+            isPortedFxCopRule: false,
+            isDataflowRule: false);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(s_Rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            context.RegisterCompilationStartAction(OnCompilationStart);
+        }
+
+        private void OnCompilationStart(CompilationStartAnalysisContext context)
+        {
+            var compilation = context.Compilation;
+            var spanType = compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemSpan1);
+
+            if (spanType == null)
+            {
+                return;
+            }
+
+            context.RegisterOperationAction(
+                operationContext =>
+                {
+                    var invocation = (IInvocationOperation)operationContext.Operation;
+
+                    if (!SymbolEqualityComparer.Default.Equals((invocation.Instance.Type as INamedTypeSymbol)?.ConstructedFrom, spanType))
+                    {
+                        return;
+                    }
+
+                    if (invocation.TargetMethod.Name != FillMethod)
+                    {
+                        return;
+                    }
+
+                    if (invocation.Arguments.Length != 1)
+                    {
+                        return;
+                    }
+
+                    if (IsDefaultValue(invocation.Arguments[0]))
+                    {
+                        operationContext.ReportDiagnostic(Diagnostic.Create(s_Rule, invocation.Syntax.GetLocation()));
+                    }
+                },
+                OperationKind.Invocation);
```

Comment:
Renamed `operationContext` to `context`.

```suggestion
                context =>
                {
                    var invocation = (IInvocationOperation)context.Operation;

                    if (!SymbolEqualityComparer.Default.Equals((invocation.Instance.Type as INamedTypeSymbol)?.ConstructedFrom, spanType))
                    {
                        return;
                    }

                    if (invocation.TargetMethod.Name != FillMethod)
                    {
                        return;
                    }

                    if (invocation.Arguments.Length != 1)
                    {
                        return;
                    }

                    if (IsDefaultValue(invocation.Arguments[0]))
                    {
                        context.ReportDiagnostic(Diagnostic.Create(s_Rule, invocation.Syntax.GetLocation()));
                    }
                },
                OperationKind.Invocation);
```